### PR TITLE
cleanup

### PR DESCRIPTION
**Pull request by hzxuzhonghu**

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
cc @ramaraochavali

File: pilot/pkg/proxy/envoy/v2/ads.go [hunk 1]
```diff
@@ -466,18 +463,10 @@ func listEqualUnordered(a []string, b []string) bool {
 
 // update the node associated with the connection, after receiving a a packet from envoy, also adds the connection
 // to the tracking map.
-func (s *DiscoveryServer) initConnection(node *core.Node, con *XdsConnection) (func(), error) {
-	con.mu.RLock() // may not be needed - once per connection, but locking for consistency.
-	initialized := con.node != nil
-	con.mu.RUnlock()
-
-	if initialized {
```
Comment (howardjohn) [inline pilot/pkg/proxy/envoy/v2/ads.go:474]: why remove this? seems like its useful

Comment (costinm) [reply pilot/pkg/proxy/envoy/v2/ads.go:474]: method is called only if con.node != nil.

Comment (costinm) [reply pilot/pkg/proxy/envoy/v2/ads.go:474]: But it would really help to have a comment.

Comment (howardjohn) [reply pilot/pkg/proxy/envoy/v2/ads.go:474]: ah right. LGTM then

File: pilot/pkg/proxy/envoy/v2/ads.go [hunk 2]
```diff
@@ -304,11 +303,9 @@ func (s *DiscoveryServer) StreamAggregatedResources(stream ads.AggregatedDiscove
 // an error is returned. For example, if a v2 cluster request was sent initially, then a v3 response was received, we will throw an error.
 // This is to ensure that when we do pushes, we are sending a consistent type, rather than flipping between v2 and v3.
 // A proper XDS client will not send mixed versions.
-func (s *DiscoveryServer) handleTypeURL(con *XdsConnection, typeURL string, requestedType *string) error {
```
Comment (costinm) [inline pilot/pkg/proxy/envoy/v2/ads.go:307]: Maybe in a future cleanup we can use the newly added model.WatchedType instead of a *string.

**Pull request merged**